src/__swaps__/screens/Swap/components/UserAssetsSync.tsx

```diff
@@ -0,0 +1,36 @@
+import { useAccountSettings } from '@/hooks';
+import { useUserAssets } from '../resources/assets';
+
+import { selectUserAssetsList, selectorFilterByUserChains } from '@/__swaps__/screens/Swap/resources/_selectors/assets';
+import { Hex } from 'viem';
+import { userAssetsStore } from '@/state/assets/userAssets';
+import { ParsedSearchAsset } from '@/__swaps__/types/assets';
+import { mergeMaps, mergeSets } from '@/__swaps__/utils/assets';
+
+export const UserAssetsSync = () => {
+  const { accountAddress: currentAddress, nativeCurrency: currentCurrency } = useAccountSettings();
+
+  useUserAssets(
+    {
+      address: currentAddress as Hex,
+      currency: currentCurrency,
+    },
+    {
+      select: data =>
+        selectorFilterByUserChains({
+          data,
+          selector: selectUserAssetsList,
+        }),
+      onSuccess: data => {
```

Comment:
this works here, but `onSuccess` is deprecated in the future versions of react query

not sure if we can run into concurrency issues if doing it inside the queryFn?

Reply:
Oh that's annoying. I use this handler all the time..